### PR TITLE
[Fix] Scope Gemma 4 image masks to sliding attention

### PR DESCRIPTION
**Pull request by BiggieW**
## Motivation
Gemma 4 expects different attention-mask semantics by layer type([code reference](https://github.com/google-deepmind/gemma/blob/bdbfebf29b453dcea5268ddfd7a8b56dc9f7061d/gemma/gm/nn/gemma4/_config.py#L129)):
- Full attention: `causal`
- Sliding attention: `(causal OR same_image_block) AND sliding_window`

SGLang applied the bidirectional same-image mask to both layer types, allowing full-attention image tokens to attend to future tokens.
## Modifications
- Added optional sliding-attention custom-mask metadata to the Triton backend.
- Updated both regular and deterministic Triton extend paths to select masks by layer type.
- Stored Gemma 4 image masks as sliding-attention overrides.
- Added CPU regression tests covering:
  - causal masking for full-attention layers
  - bidirectional same-image masking within sliding attention
  - fallback to generic custom masks
## Accuracy Tests
Added and passed:
```bash
python test/registered/unit/layers/attention/test_gemma4_attention_mask.py
```
Both regression tests pass. Full model accuracy evaluation was not run.
## Speed Tests and Profiling
Not run. This change only selects between existing mask references before kernel dispatch and does not add kernel computation or tensor allocation.
## Checklist
- [x] Format code with pre-commit.
- [x] Add unit tests.
- [x] Documentation is not required because there is no user-facing API change.
- [ ] Provide full model accuracy and speed benchmark results.
- [x] Follow the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :x: [Run #30080928548](https://github.com/sgl-project/sglang/actions/runs/30080928548)
Latest PR Test (Extra): :x: [Run #30080928464](https://github.com/sgl-project/sglang/actions/runs/30080928464)

### CODE REVIEW

**Comment by BiggieW** [comment]
Hi maintainers, thanks for taking a look at this PR! It seems the CI checks are blocked because I don't yet have permission to add the run-ci label. Would someone mind helping trigger CI for this PR when you get a chance?

Really appreciate it!

**Comment by BiggieW** [comment]
Hi @kpham-sgl @pyc96, could one of you help triage this Gemma 4 correctness fix and enable CI?

The current red Base/Extra checks are failing at `pr-gate`; the CPU/GPU test jobs have not run because I do not have permission to add the `run-ci` label.

The issue is that Gemma 4 image-token bidirectional masks are currently passed to both full- and sliding-attention layers. This PR scopes them to sliding-attention layers so that full-attention layers remain causal. The same layer-scoping direction has also been independently identified in #33033; that issue tracks the separate quadratic dense-mask performance problem.

I have added CPU regression coverage for:

* causal behavior on full-attention layers,
* same-image bidirectional behavior on sliding-attention layers,
* fallback to the generic custom mask.

I can also add a consecutive image/non-image batch test to verify that the new sliding-mask metadata cannot leak across forwards, and provide a small Gemma 4 end-to-end output/accuracy comparison if that would help review.
